pytheus/metrics.py

```diff
@@ -111,13 +120,38 @@ def __init__(
 
         # TODO: value should be threadsafe and possibly support different kinds :)
         # possibly asyncio support could be considered and also multiprocessing
-        self._value: float = 0
+        self._value: LockedValue = LockedValue()  # TODO: support multiples
 
-    def inc(self, value: float) -> None:
-        if not self._can_observe:
-            raise UnobservableMetricException
+    def inc(self, value: float = 1.0) -> None:
+        """
+        Increments the value by the given amount.
+        By default it will be 1.
+        value must be >= 0.
+        """
+        self._raise_if_cannot_observe()
+        if value < 0:
+            raise ValueError(f'Counter increase value ({value}) must be >= 0')
+
+        self._value.inc(value)
 
-        self._value += value
+    # TODO: consider adding decorator support
+    @contextmanager
+    def count_exceptions(
+        self,
+        exceptions: type[Exception] | tuple[Exception] | None = None
+    ) -> Generator[None, None, None]:
+        """
+        Will count and reraise raised exceptions.
+        It is possibly to specify which exceptions to track.
+        """
+        if exceptions is None:
+            exceptions = Exception
+
+        try:
+            yield
+        except exceptions:
```

Comment:
mypy complains here with `error: Exception type must be derived from BaseException` but I've ignored it for now